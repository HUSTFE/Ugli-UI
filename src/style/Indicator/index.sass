.ugli-spinner
  animation: ugli-spinner-rotate 0.8s infinite linear
  border: 4px solid transparent
  border-radius: 50%

@keyframes ugli-spinner-rotate
  0%
    transform: rotate(0deg)
  100%
    transform: rotate(360deg)
