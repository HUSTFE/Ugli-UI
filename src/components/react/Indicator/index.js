import React from 'react'
import style from 'style/Indicator/index.sass'

class Spinner extends React.Component {
  static defaultProps = {
    color: 'aqua',
    size: '16',
  }
  computeSize = () => `${this.props.size}px`

  render() {
    const size = this.computeSize()
    const { color } = this.props
    return (
      <div
        className={style['ugli-spinner']}
        style={{
          borderTopColor: color,
          borderLeftColor: color,
          borderBottomColor: color,
          height: size,
          width: size,
        }}
      />
    )
  }
}

export default Spinner
