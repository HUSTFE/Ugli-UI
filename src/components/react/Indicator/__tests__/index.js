import React from 'react'
import renderer from 'react-test-renderer'
import Indicator from '../index'

describe('Indicator', () => {
  it('snake spinner render without crash', () => {
    const component = renderer.create(<Indicator />)
    const json = component.toJSON()
    expect(json).toMatchSnapshot()
  })
})
