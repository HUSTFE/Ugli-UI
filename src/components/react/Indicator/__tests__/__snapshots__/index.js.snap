// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indicator snake spinner render without crash 1`] = `
<div
  className="ugli-spinner"
  style={
    Object {
      "borderBottomColor": "aqua",
      "borderLeftColor": "aqua",
      "borderTopColor": "aqua",
      "height": "16px",
      "width": "16px",
    }
  }
/>
`;
